### ISSUE TITLE
Search instance can't see provisioned Cognitive Services instances from ARM template

### ISSUE THREAD

**Issue by csiebler** [Azure-Samples/azure-search-knowledge-mining#64]
### This issue is for a: (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [x] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
1. Deploy ARM template via `Deploy to Azure` to Azure Service (location e.g. `eastus`, but same in `westeurope`, tiers all S0 or compareable)
1. Goto Search Service and select `Import Data`, pick sample dataset (or any other), hit `next`
1. Under `Attach Cognitive Services`, the provisioned `Microsoft.CognitiveServices/accounts` will not show up

Azure Cognitive Search can't find the CogServices service:
![image](https://user-images.githubusercontent.com/525867/87780839-cdccf180-c82f-11ea-88a8-70c9c1333b25.png)

But it exists in the RG:
![image](https://user-images.githubusercontent.com/525867/87780892-e76e3900-c82f-11ea-984e-0bac0a4d7117.png)

Creating a new `CognitiveServicesAllInOne` resource shows up properly in the Search instance.

### Any log messages given by the failure
No

### Expected/desired behavior
Provisioned `Microsoft.CognitiveServices/accounts` from ARM template to show up in Search Instance 

### OS and Version?
Windows 10, but should be irrelevant (all in browser)

### Versions
Irrelevant

### Mention any other details that might be useful

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by csiebler** [comment]
Looks like `Azure Cognitive Search` only lists Cognitive Services that use the `name` of a location, instead of the human-readable `displayName`.

Example: a Cognitive Service instance with location `eastus` is discovered by Search, while `East US` is not found.

The ARM template uses human-readable names, hence the problem.

**Comment by csiebler** [comment]
Fixed with PR #65
